build-system/specifications.md

```diff
@@ -0,0 +1,4606 @@
+linked into the prods and/or libraries.
+
+```
+RCS += <name> 
+RCS_<osclass> += <name>
+```
+
+Resource definition script files for all products and libraries.
+
+```
+PROD_RCS += <name> 
+PROD_RCS_<osclass> += <name>
+PROD_RCS_DEFAULT += <name>
+```
+
+Resource definition script files for all products.
+
+```
+LIB_RCS += <name> 
+LIB_RCS_<osclass> += <name>
+LIB_RCS_DEFAULT += <name>
+```
+
+Resource definition script files for all libraries.
+
+```
+<name>_RCS += <name> 
+<name>_RCS_<osclass> += <name>
+<name>_RCS_DEFAULT += <name>
+```
+
+Resource definition script files for specified product or library.
+
+### TCL libraries
+
+Definitions of the form:
+
+```
+TCLLIBNAME += <name>
+TCLINDEX += <name>
+```
+
+result in the specified tcl files being installed to the
+`$(INSTALL_LOCATION)/lib/<arch>` directory.
+
+### Java class files
+
+Java class files can be created by the javac tool into `$(INSTALL_JAVA)`
+or into the `O.Common` subdirectory, by specifying the name of the java
+class file in the Makefile. Command line options for the javac tool can
+be specified. The configuration files set the java c option
+`-sourcepath .:..:../..`.
+
+Any of the following can be specified:
+
+```
+JAVA += <name>.java
+```
+
+The `<name>.java` file will be used to create the `<name>.class` file
+in the `$(INSTALL_JAVA)` directory.
+
+```
+TESTJAVA += <name>.java
+```
+
+The `<name>.java` files will be used to create the `<name>.class` file
+in the `O.Common` subdirectory.
+
+```
+USR_JAVACFLAGS += <name>
+```
+
+The javac option `<name>` will be used on the javac command lines.
+
+#### Example 1
+
+In this example, three class files are created in
+`$(INSTALL_LOCATION)/javalib/mytest`. The javac depreciation flag is used
+to list the description of each use or override of a deprecated member
+or class.
+
+```makefile
+JAVA = mytest/one.java
+JAVA = mytest/two.java
+JAVA = mytest/three.java
+USR_JAVACFLAGS = -deprecation
+```
+
+#### Example 2
+
+In this example, the `test.class` file is created in the `O.Common`
+subdirectory.
+
+```
+TESTJAVA = test.java
+```
+
+### Java jar file
+
+A single java jar file can be created using the java jar tool and installed into
+`$(INSTALL_JAVA)` (i.e. `$(INSTALL_LOCATION)/javalib`) by specifying its name,
+and the names of its input files to be included in the created jar file. The jar
+input file names must appear with a directory prefix.
+
+Any of the following can be specified:
+
+```
+JAR += <name>
+```
+
+The `<name>` jar file will be created and installed into the
+`$(INSTALL_JAVA)` directory.
+
+```
+JAR_INPUT += <name>
+```
+
+Names of images, audio files and classes files to be included in the
+jar file.
+
+```
+JAR_MANIFEST += <name>
+```
+
+The preexisting manifest file will be used for the created jar file.
+
+```
+JAR_PACKAGES += <name>
+```
+
+Names of java packages to be installed and added to the created jar
+file.
+
+#### Example 2 1
```

Comment:
Maybe Example 2.1  or 2-1 ?

Reply:
The example names are kind of random; I switched them all to Example 1 2 3 4 5 6.